rllib/algorithms/algorithm_config.py

```diff
@@ -3343,402 +3352,199 @@ def get_evaluation_config_object(
 
         return eval_config_obj
 
-    def get_multi_agent_setup(
+    # TODO: Move this to those algorithms that really need this, which is currently
+    #  only A2C and PG.
+    def validate_train_batch_size_vs_rollout_fragment_length(self) -> None:
+        """Detects mismatches for `train_batch_size` vs `rollout_fragment_length`.
+
+        Only applicable for algorithms, whose train_batch_size should be directly
+        dependent on rollout_fragment_length (synchronous sampling, on-policy PG algos).
+
+        If rollout_fragment_length != "auto", makes sure that the product of
+        `rollout_fragment_length` x `num_env_runners` x `num_envs_per_env_runner`
+        roughly (10%) matches the provided `train_batch_size`. Otherwise, errors with
+        asking the user to set rollout_fragment_length to `auto` or to a matching
+        value.
+
+        Also, only checks this if `train_batch_size` > 0 (DDPPO sets this
+        to -1 to auto-calculate the actual batch size later).
+
+        Raises:
+            ValueError: If there is a mismatch between user provided
+                `rollout_fragment_length` and `total_train_batch_size`.
+        """
+        if (
+            self.rollout_fragment_length != "auto"
+            and not self.in_evaluation
+            and self.total_train_batch_size > 0
```

Comment:
I often saw that, if `train_batch_size` was defined, the `train_batch_size_per_learner` was `None`. The same holds true if `train_batch_size`  is not user-defined at all. Can we fill this gap?

Reply:
We should go always with `train_batch_size_per_learner` on the new stack.

The default behavior is:

if `train_batch_size_per_learner` is None, derive it from `train_batch_size` via: `train_batch_size_per_learner = train_batch_size // num_learners`.

if `train_batch_size_per_learner` is defined, use that on the new API stack (ignore on the old API stack)

Reply:
We should maybe hard-deprecate (error) completely `train_batch_size` on the new API stack.